core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerProxy.scala

```diff
@@ -391,10 +391,21 @@ class ContainerProxy(factory: (TransactionId,
       activeCount -= 1
       destroyContainer(data.container)
 
-    // Failed for a subsequent /run
-    case Event(_: FailureMessage, data: WarmedData) =>
+    case Event(failed: FailureMessage, data: WarmedData) =>
```

Comment:
If we handle as ApplicationError, this failure handling won't be needed either; ApplicationError is handled at https://github.com/apache/openwhisk/blob/ce6a5b7611c46f7d8a528c9987cfb9e9d50ade44/core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerProxy.scala#L718 to cause successful completion (container will remain in use), but errored Activation response.